src/app/webdav/blob-service.service.ts

```diff
@@ -0,0 +1,131 @@
+import { Injectable } from '@angular/core';
+import { AppConfig } from '../app-config/app-config';
+import webdav from 'webdav';
+import { DomSanitizer, SafeUrl } from '@angular/platform-browser';
+import { SessionService } from 'app/session/session.service';
+
+@Injectable()
+export class BlobService {
+
+  // TODO Check connection/login success?
+  private client: any;
+  // private defaultImage: SafeUrl;
+  private fileList: string;
+  private currentlyGettingList: Promise<boolean>;
```

Comment:
This might be a good use case for a `BehaviorSubject<boolean>` (https://www.learnrxjs.io/subjects/behaviorsubject.html)

A BehaviorSubject is basically an Observable with a default value. It's a little confusing to wrap your head around in the beginning but it can be quite useful.

EDIT: made a commit with suggestion to simplify this.

Reply:
The proposed fix currently breaks the service. I will look into it though.

Reply:
`  public async doesFileExist(name: string): Promise<boolean> {
    const fileList = await this.getFileList();`
does not work for me. Reverted the change for now.